pkg/controllers/apps/aggregatestatus/aggregatestatus.go

```diff
@@ -0,0 +1,316 @@
+/*
+Copyright 2022 The Clusternet Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aggregatestatus
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	appsapi "github.com/clusternet/clusternet/pkg/apis/apps/v1alpha1"
+	clusternetClientSet "github.com/clusternet/clusternet/pkg/generated/clientset/versioned"
+	appinformers "github.com/clusternet/clusternet/pkg/generated/informers/externalversions/apps/v1alpha1"
+	applisters "github.com/clusternet/clusternet/pkg/generated/listers/apps/v1alpha1"
+	"github.com/clusternet/clusternet/pkg/known"
+)
+
+type SyncHandlerFunc func(subscription *appsapi.Subscription) error
+
+// Controller is a controller that handle Description
+type Controller struct {
+	clusternetClient clusternetClientSet.Interface
+
+	// workqueue is a rate limited work queue. This is used to queue work to be
+	// processed instead of performing it as soon as a change happens. This
+	// means we can ensure we only process a fixed amount of resources at a
+	// time, and makes it easy to ensure we are never processing the same item
+	// simultaneously in two different workers.
+	workqueue workqueue.RateLimitingInterface
+
+	descSynced cache.InformerSynced
+	subsLister applisters.SubscriptionLister
+	subsSynced cache.InformerSynced
+
+	recorder record.EventRecorder
+
+	syncHandlerFunc SyncHandlerFunc
+}
+
+func NewController(clusternetClient clusternetClientSet.Interface,
```

Comment:
Where is the caller? 

Here you construct a new SubscriptionController. Is it better add a new queue and new synchandler to current SubscriptionController thus almost all functions could be reused.

Reply:
It's one of clusternet-hub controllers, callback `handleAggregateStatus` to sync status. They were indeed together before, but it would look messier. It is more convenient to maintain and iterate here as a single controller.